src/test/java/com/yahoo/sketches/counting/FrequentItemsTest.java

```diff
@@ -0,0 +1,157 @@
+   */
+  private long randomGeometricDist(double prob){
+    return (long) (Math.log(Math.random()) / Math.log(1.0 - prob));
+  }
+  
+  @Test
+  public void testRandomGeometricDist() {
+    long maxKey = 0L;
+    double prob = .1;
+    for (int i=0; i<100; i++){
+      // Draws a number from a skewed distribution (geometric)
+      long key = randomGeometricDist(prob) ;
+      if (key > maxKey) maxKey = key;
+      //System.out.println(key);
+      Assert.assertTrue((double)maxKey < 20.0/prob);
+    }
+  }
+   
+  @Test
+  public void realCountsInBounds() {
+    int n = 4213;
+    int maxSize = 50;
+    long key;
+    double prob = .04; 
+    FrequentItems frequentItems = new FrequentItems(maxSize);
+    PositiveCountersMap realCounts = new PositiveCountersMap();
+    for (int i=0; i<n; i++){   
+      key = randomGeometricDist(prob);
+      frequentItems.update(key);
+      realCounts.increment(key);
+      long realCount = realCounts.get(key);
+      long lowerBound = frequentItems.get(key);
+      long upperBound = frequentItems.get(key) + frequentItems.getMaxError();
+      Assert.assertTrue(upperBound >=  realCount && realCount >= lowerBound);   
+    }
+  }
+  
+  @Test
+  public void errorWithinLimits() {
+    int n = 100;
+    int maxSize = 20;
+    long key;
+    double prob = .1;
+    FrequentItems frequentItems = new FrequentItems(maxSize);
+    for (int i=0; i<n; i++){
+      key = randomGeometricDist(prob);
+      frequentItems.update(key);
+      long lowerBound = frequentItems.get(key);
+      long upperBound = frequentItems.get(key) + frequentItems.getMaxError();
+      Assert.assertTrue(upperBound - lowerBound <= i/maxSize);  
+    }
+  } 
+    
+  @Test
+  public void realCountsInBoundsAfterUnion() {
+    int n = 1000;
+    int maxSize1 = 100;
+    int maxSize2 = 400;
+    double prob1 = .01;
+    double prob2 = .005;
+   
+    PositiveCountersMap realCounts = new PositiveCountersMap();
+    FrequentItems frequentItems1 = new FrequentItems(maxSize1);
+    FrequentItems frequentItems2 = new FrequentItems(maxSize2);
+    for (int i=0; i<n; i++){
+      long key1 = randomGeometricDist(prob1);
+      long key2 = randomGeometricDist(prob2);
+      
+      frequentItems1.update(key1);
+      frequentItems2.update(key2);
+      
+      // Updating the real counters
+      realCounts.increment(key1);
+      realCounts.increment(key2);
+    }
+    FrequentItems frequentItems = frequentItems1.union(frequentItems2);
+
+    for ( long key : realCounts.keys()){
+      
+      long realCount = realCounts.get(key);
+      long lowerBound = frequentItems.get(key);
+      long upperBound = frequentItems.get(key) + frequentItems.getMaxError();
+      Assert.assertTrue(upperBound >=  realCount && realCount >= lowerBound);
+    }
+  }
+  
+  @Test
+  public void stressTestUpdateTime() {
+    int n = 1000000;
+    int maxSize = 10000;  
+    FrequentItems frequentItems = new FrequentItems(maxSize);
+    double prob = 1.0/((double) n);
```

Comment:
Either eliminate the unnecessary cast or add a @SuppressWarnings("cast") to the method.